src/board.py

```diff
@@ -127,16 +148,49 @@ def __contains__(self, piece: Piece | None) -> bool:
         """
         return any(piece in rank for rank in self._board)
 
-    def square_of(self, piece: Piece | None) -> Square | None:
-        """Return the square of a specific piece.
+    @property
+    def moves(self):
+        """Generate moves for all pieces on the board."""
+        _moves = {}
 
-        Args:
-            piece: Piece object to fetch the square of.
+        for piece in self:
+            def deployable(source_piece: Piece, target: Square):
+                Piece.deployable.__doc__
+                target_piece = self[target]
 
-        Returns:
-            The square of the piece in chess notation or indices or None if it is not on the board.
+                return Piece.deployable(source_piece, target) and target_piece is None
+
+            def capturable(source_piece: Piece, target: Square):
+                Piece.capturable.__doc__
+                target_piece = self[target]
+
+                if target_piece is not None:
+                    return Piece.capturable(source_piece, target) and source_piece.orientation != target_piece.orientation
+
+                return False
+
+            piece.deployable = MethodType(deployable, piece)
```

Comment:
Some of these checks are simply because Pylance complains. With the use of invariants like you say in another remark, I can replace several of these checks with the `# type: ignore` Pylance flag.

I really thought of the `Empty` piece at some point, and maybe that of the empty square even, but I would like to avoid it for the moment. Lets see how clean it becomes with the use of `None` for empty, and work from there.

Reply:
I did not go through with the empty implementation.